File: java/src/main/java/shapes/base/commons/Displayable.java
package shapes.base.commons;

public interface Displayable {
    default boolean shouldDisplay() {
        return true;
    }

    default void display() {
        if (shouldDisplay()) {
            System.out.println("Yes, this shape can be displayed.");
        } else {
            System.out.println("No, this shape cannot be displayed.");
        }
    }
}

File: scala/src/main/scala/shapes/base/commons/Displayable.scala
package shapes.base.commons

trait Displayable {
    def shouldDisplay: Boolean = {
        true
    }

    def display(): Unit = {
        if (shouldDisplay) {
            println("Yes, this shape can be displayed.")
        } else {
            println("No, this shape cannot be displayed.")
        }
    }
}
